Introducing Configuration transporter

pkamps opened the pull request:
Currently, it's only possible to syncronize the elevate configuration with a solr instance directly. At a client installation, we do have a different setup: the solr configuration is managed by zookeeper - a system to have high availability for solr.

This pull request introduces "Configuration transporter" classes. There is one for "solr" executing the existing syncronization and a new one for zookeeper. There is a factory function to chose a transporter based on the settings.

pkamps committed:
Add new configuration transporter classes

The commit introduces two new configuration transporter classes for Solr and Zookeeper. It also moves the configuration synchronization method away from the eZSolr class into the new classes with a factory pattern implementation to instantiate the optimal transporter.

pkamps committed:
Refactor Solr configuration sync and error handling

Moving 2 functions from ezfindelevateconfiguration.php into the newly created transporter class.

benkmugo commented:
Comments:
In eZFConfigurationTransporterZookeeperCli->push() I think it would make sense read the path from a config parameter like you're doing with the CLI already.
Also with a fallback like in case of the CLI. That way it is configurable via INI.

Questions:
What would be the easiest way to test the non-zookeeper transporter?
Install in Lovestack and use the elevate UI in the admin?

pkamps committed:
Add configurable file path for Zookeeper CLI configuration transporter

The changes allow flexibility when setting the configuration file path for the eZFConfigurationTransporterZookeeperCli class. Previously, the path was hardcoded into the class, but now it can be set via ConfigurationTransporterParameters in solr.ini.

pkamps commented:
You can now configure the path to the config file in the settings.

Answers are in a separate email.

pkamps committed:
Reloading the solr collection after updating the elevate.xml configuration inside the zookeeper repo

pkamps committed:
Handle elevate updates whenever a content objects gets removed
